src/components/expression/expression.js

```diff
@@ -172,6 +204,22 @@ genotet.ExpressionView = function(viewName, params) {
     .on('genotet.updatePanel', function() {
       this.panel.dataLoaded();
     }.bind(this));
+
+  // Set up link callbacks.
+  $(this).on('genotet.addProfile', function(event, data) {
+    var genes = data.id.split(',');
```

Comment:
How is the data.id a comma separated string in the first place? You can convert it to an array of ids before signaling.